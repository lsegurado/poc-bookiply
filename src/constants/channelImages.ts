import AIRBNB from '../assets/AIRBNB.svg';
import BOOKINGCOM from '../assets/BOOKINGCOM.svg';
import HOLIDU from '../assets/HOLIDU.svg';
import { ChannelType } from '../types';

export const channelImages: Record<ChannelType, string> = {
    AIRBNB,
    BOOKINGCOM,
    HOLIDU
};
