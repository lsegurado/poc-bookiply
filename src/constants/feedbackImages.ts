import thumbUp from '../assets/thumb-up.svg';
import thumbDown from '../assets/thumb-down.svg';

/**Images related to positive / negative feedback */
export const feedbackImages = {
    thumbUp,
    thumbDown
}