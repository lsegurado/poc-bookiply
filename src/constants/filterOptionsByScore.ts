/**Predefined options for filtering by score */
export const filterOptionsByScore = {
    /**Minimum score to look for */
    availableScoringFilters: [4.5, 4.0, 3.5, 3.0],
    /**Maximum score */
    maximumScore: 5,
    /**Precision used to send the filter by scores to the API */
    filterAccuracy: 0.1,
    /**Accuracy to be displayed by the stars in the filter */
    starsPrecision: 0.5
};
