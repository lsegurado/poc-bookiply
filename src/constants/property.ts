import background from '../assets/background.png';

export const property = {
    name: 'La Casa de las Flores',
    id: '091021',
    background
}