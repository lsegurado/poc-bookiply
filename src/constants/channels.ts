import { ChannelType } from "../types";
import { channelImages } from "./channelImages";

/**A channel list */
export const channels = Object.keys(channelImages) as ChannelType[]