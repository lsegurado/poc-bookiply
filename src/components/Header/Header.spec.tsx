import { render, RenderResult } from "@testing-library/react";
import { Header } from ".";
import { property } from "../../constants/property";

const { id, name, background } = property;
describe('Header component', () => {
    let el: RenderResult;
    beforeEach(() => {
        el = render(<Header />);
    })
    it(`should match snapshot`, () => {
        expect(el.container).toMatchSnapshot();
    })
    it(`should render a header container`, () => {
        expect(el.getByRole('banner').tagName).toEqual('HEADER');
    })
    it(`should render id`, () => {
        expect(el.getByText(`ID: ${id}`)).toBeDefined();
    })
    it(`should render title`, () => {
        expect(el.getByText(name)).toBeDefined();
    })
    it(`should render background`, () => {
        expect(el.getByRole('img')).toBeDefined();
        expect(el.getByRole('img').getAttribute('src')).toEqual(background);
    })
});