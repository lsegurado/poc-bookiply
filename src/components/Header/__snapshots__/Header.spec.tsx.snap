// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should match snapshot 1`] = `
<div>
  <header>
    <p
      class="id"
    >
      ID: 
      091021
    </p>
    <h1
      class="title"
    >
      La Casa de las Flores
    </h1>
    <img
      alt="La Casa de las Flores"
      class="img"
      src="background.png"
    />
    <div
      class="fader"
    />
  </header>
</div>
`;
